src/iosrv/sched.c

```diff
@@ -0,0 +1,539 @@
+/**
+ * (C) Copyright 2016-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#define D_LOGFAC       DD_FAC(server)
+
+#include <abt.h>
+#include <daos/common.h>
+#include <daos_errno.h>
+#include "srv_internal.h"
+
+/*
+ * A schedule cycle consists of three stages:
+ * 1. Starting with a network poll ULT, number of ULTs to be executed in this
+ *    cycle is queried by ABT_pool_get_size() for each non-poll ABT pool;
+ * 2. Executing all the non-poll ULTs;
```

Comment:
what is non-poll ULTs?

Reply:
All other ULTs except of network poll & NVMe poll.